### PR TITLE
progress

### PR DESCRIPTION
**Pull request by mejango**

### CODE REVIEW

**Review by viraj124 — APPROVED** [review]

File: contracts/JBTieredLimitedNFTRewardDataSource.sol [hunk 1]
```diff
@@ -190,7 +190,7 @@ contract JBTieredLimitedNFTRewardDataSource is
     IJBDirectory _directory,
     string memory _name,
     string memory _symbol,
-    IToken721UriResolver _tokenUriResolver,
+    IJBTokenUriResolver _tokenUriResolver,
```
Comment (drgorillamd) [inline contracts/JBTieredLimitedNFTRewardDataSource.sol:193]: Isn't this a bit misleading with the erc20 JBToken?

File: contracts/abstract/JBNFTRewardDataSource.sol [hunk 2]
```diff
@@ -179,62 +238,22 @@ abstract contract JBNFTRewardDataSource is
...
-      string memory memo,
-      IJBPayDelegate delegate
-    )
-  {
-    // Make sure the caller is expected, and the call is being made on behalf of an interaction with the correct project.
+  function didPay(JBDidPayData calldata _data) external override {
+    // Make sure the caller is a terminal of the project, and the call is being made on behalf of an interaction with the correct project.
     if (
-      !directory.isTerminalOf(projectId, _data.terminal) ||
-      msg.sender != _expectedCaller ||
```
Comment (drgorillamd) [inline contracts/abstract/JBNFTRewardDataSource.sol:205]: :(

File: contracts/abstract/JBNFTRewardDataSource.sol [hunk 3]
```diff
@@ -179,62 +238,22 @@ abstract contract JBNFTRewardDataSource is
...
+      !directory.isTerminalOf(projectId, IJBPaymentTerminal(msg.sender)) ||
```
Comment (drgorillamd) [inline contracts/abstract/JBNFTRewardDataSource.sol:251]: msg.sender is the terminal store!

Comment (mejango) [reply contracts/abstract/JBNFTRewardDataSource.sol:251]: not for the delegate! The delegate gets called from the terminal proper.

Comment (drgorillamd) [reply contracts/abstract/JBNFTRewardDataSource.sol:251]: but this is payParams, isn't it the dataSource?

Comment (drgorillamd) [reply contracts/abstract/JBNFTRewardDataSource.sol:251]: oh no, mb, this became didPay, nvm

**Pull request merged**